obspy/core/util/version.py

```diff
@@ -195,5 +195,82 @@ def _normalize_version(version):
     return version
 
 
+def cmp_obspy_version(x, y):
+    """
+    Compare two ObsPy version number strings.
+
+    Similar to python builtin :py:func:`cmp()`.
+    The return value is negative if ``x < y``, zero if ``x == y`` and
+    strictly positive if ``x > y``.
+
+    The return type is ``int`` for comparisons that only involve well defined
+    release (or release candidate) versions. The return type is ``float`` for
+    comparisons that involve at least one development version.
+    In case of two development versions based on the same point release (e.g.
+    "1.0.1.post0+62.g3718497307.dirty" and "1.0.1.post0+55.gcdc54dbd9a")
+    the return value is also zero (type ``float``).
+
+    >>> a = "1.0.0"
+    >>> b = "1.0.1"
+    >>> c = "1.0.1.post0+62.g3718497307.dirty"
+    >>> d = "1.0.1.post0+55.gcdc54dbd9a"
+    >>> cmp_obspy_version(a, b)
+    -1
+    >>> cmp_obspy_version(a, c)
+    -1.0
+    >>> cmp_obspy_version(b, c)
+    -1.0
+    >>> cmp_obspy_version(b, b)
+    0
+    >>> cmp_obspy_version(c, d)
+    0.0
+
+    :type x: str
+    :type y: str
+    :rtype: int or float
+    """
+    def _version_tuple(version_string):
+        """
+        Convert string representation of ObsPy version into a tuple for
+        comparison.
+
+        >>> version = "1.0.1"
+        >>> _version_tuple(version)
+        (1, 0, 1)
+        >>> version = "1.0.1.post0+135.gf5ad840dce"
+        >>> _version_tuple(version)
+        (1, 0, 1, u'post0+135.gf5ad840dce')
+        """
+        return tuple(map(int, version_string.split(".", 3)[:3]) +
+                     version_string.split(".", 3)[3:])
+    x = _version_tuple(x)
+    y = _version_tuple(y)
+
+    # when comparing two development versions: only compare the point release
+    # both are based on
+    if len(x) == 4 and len(y) == 4:
+        _x = x[:3]
+        _y = y[:3]
```

Comment:
Maybe even a namedtuple with `major`, `minor`, `micro`, and `dev` fields. Python has this:
```python
In [4]: sys.version_info
Out[4]: sys.version_info(major=3, minor=5, micro=1, releaselevel='final', serial=0)
```